### PR TITLE
Decouple User Model from Doctrine ArrayCollection

### PR DESCRIPTION
**Pull request by bitgandtter**
from #1942

### CODE REVIEW

**Comment by XWB** [comment]
This is not the right approach, with `ArrayCollection` you can do stuff like `$this->groups->add()` etc.

Perhaps we should do something similar like `FOSCommentBundle`.

https://github.com/FriendsOfSymfony/FOSCommentBundle

**Comment by bitgandtter** [comment]
Got it my bad. I should write a test for it too, i was under the assumption that a unit test was available for that. 

The main idea is to decouple Doctrine from your main architecture model. 

When you say like FOSCommentBundle on which section do you recommend to look?

File: Model/User.php [hunk 1]
```diff
@@ -88,7 +85,7 @@
     protected $passwordRequestedAt;
 
     /**
-     * @var Collection
+     * @var Array
```
Comment (stof) [inline Model/User.php]: this cannot be an array. Doctrine will put a collection in it when loading the object.

btw, ``array`` should be lowercased.

File: Model/User.php [hunk 2]
```diff
@@ -199,7 +196,7 @@ public function unserialize($serialized)
             $this->credentialsExpireAt,
             $this->email,
             $this->emailCanonical
-        ) = $data;
+            ) = $data;
```
Comment (stof) [inline Model/User.php]: code formatting should not be changed to different standards.

**Comment by stof** [comment]
This cannot be merged because it is broken. The entity will contain a collection, not an array.

**Comment by bitgandtter** [comment]
Sorry about all the inconvenients. 

Maybe the rigth way is just to remove the auto creation of the groups array and entrust developers through doc to implement their own  I recognize that will break BC. 

Another approach can be move that functionality to an extended groupable class that developers can extend of too if they want, leaving the choise of the developers to extend from the right one on their arch. 

Again my main concern is to couple my whole architecture to the doctrine library.

**Pull request closed without merging**